### ISSUE TITLE
Check Mode ("Dry Run") fails in pkg-debian.yml because the deb repo hasn't been added to sources

### ISSUE THREAD

**Issue by DownRangeDevOps** [DataDog/ansible-datadog#92]
Hi, thanks for creating the role! This is very straightforward to fix, but I'm not sure which approach you would prefer. I'm happy to open a PR, just let me know what route you would like to take.

## Issue:

If you have never run `pkg-debian.yml` the system will not have the datadog deb repo added to sources. This causes this task to fail as the package cannot be found.

**Version:** `1.4.0` 
**File:** `Datadog.datadog/tasks/pkg-debian.yml:40`
**Error:** 
```
TASK [Datadog.datadog : Ensure Datadog agent is installed] *********************
fatal: [10.1.2.220]: FAILED! => {"changed": false, "failed": true, "msg": "No package matching 'dat
adog-agent' is available"}
```

## Resolution:

Two ways to fix this. 

1. Add `when: not ansible_check_mode` conditional to this step so that it is skipped when running `ansible-playbook --check`
2. Add `check_mode: no` to the prior tasks that set up the apt-key and source for the datadog deb repo which will force the key/deb repo to be added to sources on the system when running with `--check`

I usually go for option two on my systems because I don't care about a few keys being added even if not yet being used and would prefer to let the `apt` module be dry-run for testing. However, I suspect that with a public role such as this that people may want to avoid adding the key cruft so option one may be better.

Let me know, thanks!

**Comment by hush-hush** [comment]
Hi @techFishIO,

Thanks for raising this issue !
I will look into it, seems to me that solution 1 would be better indeed.

**Comment by hush-hush** [comment]
PR #128 should fix this issue. Happy to know what you think of it if you want to give it a quick review.

**Comment by hush-hush** [comment]
#128 was merged and release with version `2.1.0` ansible role.

Thanks for opening this issue !
